src/gpu/fft.jl

```diff
@@ -0,0 +1,29 @@
+# Explicit GPU kernels for the remapping of data in K-point specific FFTs, using
+# KernelAbstractions to remain vendor agnostic. Constructs such as 
+# f_real[Gvec_mapping] = f_fourier or f_fourier .= view(f_real, Gvec_mapping)
+# are not optimized on the GPU, and performance suffers when large numbers of
+# FFT calls take place (tens of thousands or more).
+
+import KernelAbstractions: get_backend, @kernel, @index, @Const
+
+# Mapping kernel for the K-point specific iFFT
+@kernel function ifft_remap_kernel!(f_real, @Const(Gvec_mapping), @Const(f_fourier))
+    i = @index(Global)
+    @inbounds f_real[Gvec_mapping[i]] = f_fourier[i]
```

Comment:
I'm wondering if the main performance problem was actually that we did not put an `@inbounds` in the assignment `f_real[Gvec_mapping] = f_fourier`.

@abussy Can you check that the original DFTK code with a simple `@inbounds f_real[Gvec_mapping] = f_fourier` not already does the trick?

In any case if we make the inbounds assumption for the GPU version, we should do it for the CPU version, too.

Reply:
Very good point. I had a look into the impact of using `@inbounds`, and it is significant. As a matter of fact, it matches the performance of the custom kernel, so there is no need to add the `KernelAbstractions.jl` dependency, nor the `src/gpu/fft.jl` file.

To fully match the performance of the PR in its current state, it is not enough simply add `@inbounds` to the master branch code. Profiling reveals that there is still a lot of time spent transferring the `Gvec_mapping` array from the CPU to the device at each FFT call.

I would propose the keep a version of `Gvec_mapping` on the GPU, but get rid of all the kernel code.